### PR TITLE
Textfield: Setup components

### PR DESCRIPTION
**Pull request by gtchantouria**
## Useful links
- [Contributing](https://github.com/mercadolibre/fury_andesui-android/blob/develop/CONTRIBUTING.md) has more information and tips for a great pull request. This will give you a nice introduction about how to contribute with Andes.
- [Wiki](https://github.com/mercadolibre/fury_andesui-android/wiki) has all the info you need to start with Andes. Even I heard there's a video there...

## Motivation
Setup components: 
- Label
- Placeholder (Hint)
- Helper
- Counter (WIP)

## Screenshots
image

## Checks
Are you sure you followed all those steps? In such case, let's say with me "I solemnly declare that ...":
- [ ] I have coded an example inside the Demo App,
- [ ] I have added proper tests,
- [ ] I have modified the [Changelog](https://github.com/mercadolibre/fury_andesui-android/blob/develop/CHANGELOG.md) file,
- [ ] I have updated GitHub wiki,
- [ ] I have the explicit approval of one or more members of the UX Team,
- [ ] I have updated the version in gradle.properties file.

### CODE REVIEW

**Comment by furyGorgoryAdmin** [comment]
### Informative Menu <h2>  ### 

### Use any of these tags in a pull request description or leave a comment ### 
| Tag | Description |
|---|---|
| **[canary]**| Creates a build that can only be deployed with canary strategy. | 
| **[build]** |  Creates version or build from a branch that has been tagged as "no releasable". | 
| **[name:version-name]** |Specify the name of the build. For example: [build] [name:1.2.3-name] or [canary] [name:1.2.3-name].|
| **[major]** | To automatically increase the first position of the version. For example: [build] [major] or [canary] [major].|
| **[minor]** | To automatically increase the second position of the version. For example: [build] [minor] or [canary] [minor]. |
| **[patch]** | To automatically increase the third position of the version. For example: [build] [patch] or [canary] [patch].|  

[For more information about ci-cd](http://docs.furycloud.io/?lang=esp&section=businessassurance)
[For more information about semver](https://semver.org/#summary)

File: components/src/main/java/com/mercadolibre/android/andesui/textfield/AndesTextfield.kt [hunk 1]
```diff
@@ -155,4 +177,75 @@ class AndesTextfield : ConstraintLayout {
...
+
+        textComponent.addTextChangedListener(object: TextWatcher {
+            override fun afterTextChanged(s: Editable?) {
+            }
+
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+            }
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                counterComponent.text = "${count}/${config.counterMaxLength}"
```
Comment (gtchantouria) [inline components/src/main/java/com/mercadolibre/android/andesui/textfield/AndesTextfield.kt]: Lint me está puteando por esto, así que ahora lo cambio

File: components/src/main/java/com/mercadolibre/android/andesui/textfield/state/AndesTextfieldStateInterface.kt [hunk 2]
```diff
@@ -17,28 +20,38 @@ import com.mercadolibre.android.andesui.utils.buildColoredCircularShapeWithIconD
  *
  */
 internal sealed class AndesTextfieldStateInterface {
+    abstract fun primaryColor() : AndesColor
```
Comment (lautimfernandez) [inline components/src/main/java/com/mercadolibre/android/andesui/textfield/state/AndesTextfieldStateInterface.kt]: no se si primary color es un nombre correcto o que se entienda (no entiendo bien a simple vista cual es el primary) , capaz es preferible no reutilizar esta fun y dejar el color explicito en cada uno de los que lo usa

File: components/src/main/java/com/mercadolibre/android/andesui/textfield/state/AndesTextfieldStateInterface.kt [hunk 3]
```diff
@@ -17,28 +20,38 @@ import com.mercadolibre.android.andesui.utils.buildColoredCircularShapeWithIconD
...
 
 internal object AndesEnabledTexfieldState : AndesTextfieldStateInterface() {
+    override fun primaryColor(): AndesColor = R.color.andes_gray_450.toAndesColor()
+    override fun hintColor(): Int = R.color.andes_gray_450
+    override fun typeFace(context: Context): Typeface = context.getFontOrDefault(R.font.andes_font_regular)
+
     override fun backgroundColor(context: Context): Drawable {
         return StateListDrawable().apply {
             addState(intArrayOf(android.R.attr.state_focused), createGradientDrawable(context, context.resources.getDimension(R.dimen.andes_textfield_focused_stroke).toInt(), ContextCompat.getColor(context, R.color.andes_accent_color_500), null))
             addState(intArrayOf(android.R.attr.state_enabled), createGradientDrawable(context, context.resources.getDimension(R.dimen.andes_textfield_simple_stroke).toInt(), ContextCompat.getColor(context, R.color.andes_gray_200), null))
```
Comment (lautimfernandez) [inline components/src/main/java/com/mercadolibre/android/andesui/textfield/state/AndesTextfieldStateInterface.kt]: si te animas a cambiar los context compat que quedaron por R.color.andes_color_.toColor(context)

File: components/src/main/java/com/mercadolibre/android/andesui/textfield/AndesTextfield.kt [hunk 4]
```diff
@@ -15,39 +19,53 @@ import com.mercadolibre.android.andesui.textfield.state.AndesTextfieldState
...
+            setupCounterComponent(createConfig())
         }
 
+    /**
+     * Getter and setter for the state of [EditText].
+     */
     var state : AndesTextfieldState
         get() = andesTextfieldAttrs.state
         set(value) {
             andesTextfieldAttrs = andesTextfieldAttrs.copy(state = value)
```
Comment (lautimfernandez) [inline components/src/main/java/com/mercadolibre/android/andesui/textfield/AndesTextfield.kt]: Si podes hacete el set up de los componentes correspondientes cuando se cambia de estado, que serian el backgroudn, counter,  helper, placeholder , label y icon. Podriamos enmascarar estos en un setupColorComponents o algo asi y que se llame a esa fun.

**Pull request merged**